pkg/controller/kogitodataindex/resource/kafka.go

```diff
@@ -14,19 +14,153 @@
 
 package resource
 
-import "github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+import (
+	"fmt"
 
-func fromKafkaToStringMap(kafka v1alpha1.KafkaConnectionProperties) map[string]string {
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/kafka/v1beta1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	kafkaClusterLabel string = "strimzi.io/cluster"
+
+	kafkaURINotFoundError string = "there's no Kafka instance URI found in the namespace and Kafka external URI is not specified, cannot deploy Data Index service"
+	kafkaNotExist         string = "there is no Kafka instance found in the namespace, cannot deploy Kafka Topics"
+)
+
+func fromKafkaToStringMap(externalURI string) map[string]string {
 	propsmap := map[string]string{}
-	if &kafka == nil {
-		return propsmap
+	if len(externalURI) > 0 {
+		for _, envKey := range managedKafkaKeys {
+			propsmap[envKey] = externalURI
+		}
 	}
+	return propsmap
+}
 
-	if len(kafka.ServiceURI) > 0 {
-		for _, envKey := range managedKafkaKeys {
-			propsmap[envKey] = kafka.ServiceURI
+// IsKafkaServerURIResolved checks if the URI of the Kafka server is provided or resolvable in the namespace
+func IsKafkaServerURIResolved(instance *v1alpha1.KogitoDataIndex, client *client.Client) (bool, error) {
+	if len(instance.Spec.Kafka.ExternalURI) == 0 {
+		if !client.HasServerGroup("kafka.strimzi.io") {
+			return false, nil
+		}
+		if kafka, err := getKafkaInstance(instance.Spec.Kafka, instance.Namespace, client); err != nil {
+			return false, err
+		} else if kafka == nil {
+			return false, nil
 		}
 	}
+	return true, nil
+}
 
-	return propsmap
+func getKafkaServerURI(kafkaProp v1alpha1.KafkaConnectionProperties, namespace string, client *client.Client) (string, error) {
+	if len(kafkaProp.ExternalURI) > 0 {
+		return kafkaProp.ExternalURI, nil
+	} else if client.HasServerGroup("kafka.strimzi.io") {
+		if kafka, err := getKafkaInstance(kafkaProp, namespace, client); err != nil {
+			return "", err
+		} else if kafka != nil {
+			if uri := resolveKafkaServerURI(kafka); len(uri) > 0 {
+				return uri, nil
+			}
+		}
+	}
+	return "", fmt.Errorf(kafkaURINotFoundError)
+}
+
+func getKafkaServerReplicas(kafkaProp v1alpha1.KafkaConnectionProperties, namespace string, client *client.Client) (string, int32, error) {
+	if len(kafkaProp.ExternalURI) <= 0 && client.HasServerGroup("kafka.strimzi.io") {
```

Comment:
Instead of calling `HasServerGroup`, why not a function `isKafkaAvailable`? Then we won't need to change the group name every where if this changes.

Reply:
done